src/components/Breadcrumb/Breadcrumb.ts

```diff
@@ -0,0 +1,235 @@
+          }
+          a.textContent = item.text;
+          chevron.className = "ms-Breadcrumb-chevron ms-Icon ms-Icon--chevronRight";
+          listItem.appendChild(a);
+          listItem.appendChild(chevron);
+          this._breadcrumbList.appendChild(listItem);
+        }
+      }
+    }
+
+    /**
+     * resets a list by removing its children
+     */
+    private _resetList(list) {
+      while (list.firstChild) {
+        list.removeChild(list.firstChild);
+      }
+    }
+
+    /**
+     * opens the overflow menu
+     */
+    private _openOverflow(event) {
+      if (this._overflowMenu.className.indexOf(" is-open") === -1) {
+        this._overflowMenu.className += " is-open";
+        this.removeOutlinesOnClick(event);
+        // force focus rect onto overflow button
+        this._overflowButton.focus();
+      }
+    }
+
+    private _overflowKeyPress(event) {
+      if (event.keyCode === 13) {
+        this._openOverflow(event);
+      }
+    }
+
+    /**
+     * closes the overflow menu
+     */
+    private _closeOverflow(event) {
+      if (!event || event.target !== this._overflowButton) {
+        this._removeClass(this._overflowMenu, " is-open");
+      }
+    }
+
+    /**
+     * utility that removes a class from an element
+     */
+    private _removeClass(element, value) {
+      let index = element.className.indexOf(value);
+      if (index > -1) {
+        element.className = element.className.substring(0, index);
+      }
+    }
+
+    /**
+     * caches elements and values of the component
+     */
+    private _cacheDOM() {
+      this._breadcrumb = this.container;
+      this._breadcrumbList = <HTMLElement>this._breadcrumb.querySelector(".ms-Breadcrumb-list");
+      this._listItems = <NodeListOf<HTMLElement>>this._breadcrumb.querySelectorAll(".ms-Breadcrumb-listItem");
+      this._contextMenu = <HTMLElement>this._breadcrumb.querySelector(".ms-ContextualMenu");
+      this._overflowButton = <HTMLElement>this._breadcrumb.querySelector(".ms-Breadcrumb-overflowButton");
+      this._overflowMenu = <HTMLElement>this._breadcrumb.querySelector(".ms-Breadcrumb-overflowMenu");
+    }
+
+    /**
+     * sets handlers for resize and button click events
+     */
+    private _setListeners() {
+      window.addEventListener("resize", this._onResize.bind(this));
+      this._overflowButton.addEventListener("click", this._openOverflow.bind(this), false);
+      this._overflowButton.addEventListener("keypress", this._overflowKeyPress.bind(this), false);
+      document.addEventListener("click", this._closeOverflow.bind(this), false);
+      this._breadcrumbList.addEventListener("click", this.removeOutlinesOnClick, false);
+    }
+
+    /**
+     *
+     * @param {HTMLElement} container - the target container for an instance of Breadcrumb
+     * @constructor
+     *
+     * If dynamically populating a list run the constructor after the list has been populated
+     * in the DOM.
+     */
+    constructor(container: any) {
```

Comment:
Move the constructor to the top, just after the declaration of the class variables.